### PR TITLE
Get akoo package input from addon secret during cluster upgrade

### PR DESCRIPTION
**Pull request by 12345lcr**
### What this PR does / why we need it
Get akoo package input from addon secret during cluster upgrade. Since the akoo is deployed from tkg-pkg after cluster upgrade and the old akoo packageInstall will be removed, we are reusing the old akoo's addon secret for generating the akoo package input data values.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Added ut for fetching ako-operator addon secret and passed locally.

Created a tkg 1.6.0 env and manually built tanzu cli & management-package repo to test the upgrade. management-package upgrade succeeded and both the management cluster and workload cluster upgraded passed

```
~$ k get pkgi -A
NAMESPACE    NAME                                PACKAGE NAME                                         PACKAGE VERSION                            DESCRIPTION           AGE
tkg-system   ako-operator                        ako-operator-v2.tanzu.vmware.com                     0.77.0-dev                                 Reconcile succeeded   5m56s
tkg-system   antrea                              antrea.tanzu.vmware.com                              1.7.1+vmware.1-tkg.1-advanced-zshippable   Reconcile succeeded   59m
tkg-system   load-balancer-and-ingress-service   load-balancer-and-ingress-service.tanzu.vmware.com   1.8.1+vmware.1-tkg.1-zshippable            Reconcile succeeded   58m
tkg-system   metrics-server                      metrics-server.tanzu.vmware.com                      0.6.1+vmware.1-tkg.3-zshippable            Reconcile succeeded   59m
tkg-system   secretgen-controller                secretgen-controller.tanzu.vmware.com                0.11.0+vmware.2-tkg.1-zshippable           Reconcile succeeded   59m
tkg-system   tanzu-addons-manager                addons-manager.tanzu.vmware.com                      0.77.0-dev                                 Reconcile succeeded   5m33s
tkg-system   tanzu-auth                          tanzu-auth.tanzu.vmware.com                          0.77.0-dev                                 Reconcile succeeded   5m33s
tkg-system   tanzu-cliplugins                    cliplugins.tanzu.vmware.com                          0.77.0-dev                                 Reconcile succeeded   5m33s
tkg-system   tanzu-core-management-plugins       core-management-plugins.tanzu.vmware.com             0.77.0-dev                                 Reconcile succeeded   62m
tkg-system   tanzu-featuregates                  featuregates.tanzu.vmware.com                        0.77.0-dev                                 Reconcile succeeded   62m
tkg-system   tanzu-framework                     framework.tanzu.vmware.com                           0.77.0-dev                                 Reconcile succeeded   5m56s
tkg-system   tkg-clusterclass                    tkg-clusterclass.tanzu.vmware.com                    0.77.0-dev                                 Reconcile succeeded   5m56s
tkg-system   tkg-clusterclass-vsphere            tkg-clusterclass-vsphere.tanzu.vmware.com            0.77.0-dev                                 Reconcile succeeded   5m48s
tkg-system   tkg-pkg                             tkg.tanzu.vmware.com                                 0.77.0-dev                                 Reconcile succeeded   6m4s
tkg-system   tkr-service                         tkr-service.tanzu.vmware.com                         0.77.0-dev                                 Reconcile succeeded   5m33s
tkg-system   tkr-source-controller               tkr-source-controller.tanzu.vmware.com               0.77.0-dev                                 Reconcile succeeded   4m28s
tkg-system   tkr-vsphere-resolver                tkr-vsphere-resolver.tanzu.vmware.com                0.77.0-dev                                 Reconcile succeeded   4m28s
tkg-system   vsphere-cpi                         vsphere-cpi.tanzu.vmware.com                         1.24.0+vmware.1-tkg.1-zshippable           Reconcile succeeded   59m
tkg-system   vsphere-csi                         vsphere-csi.tanzu.vmware.com                         2.6.2+vmware.2-tkg.1-zshippable            Reconcile succeeded   59m
```

### Release note

```release-note

```

### CODE REVIEW

**Review by XudongLiuHarold — APPROVED** [review]
Generally LGTM, thanks @12345lcr !

File: tkg/client/management_components.go [hunk 1]
```diff
@@ -501,3 +507,43 @@ func ProcessAKOPackageInstallFile(akoPackageInstallTemplateDir, userConfigValues
 
 	return akoPackageInstallFile, nil
 }
+
+func RetrieveAKOOVariablesFromAddonSecret(clusterClient clusterclient.Client, clusterName string, configValues map[string]interface{}) error {
+	akoOperatorAddonName := fmt.Sprintf("%s-%s-addon", clusterName, constants.AkoOperatorName)
+	pollOptions := &clusterclient.PollOptions{Interval: clusterclient.CheckResourceInterval, Timeout: 3 * clusterclient.CheckResourceInterval}
+	bytes, err := clusterClient.GetSecretValue(akoOperatorAddonName, "values.yaml", constants.TkgNamespace, pollOptions)
+	if err != nil && apierrors.IsNotFound(err) {
+		log.V(6).Infof("akoo addon secret %s/%s not found, akoo was not installed on this legacy management cluster", constants.TkgNamespace, akoOperatorAddonName)
```
[the remark this answers is not in the document]
Comment (12345lcr) [reply tkg/client/management_components.go:525]: Thanks Xudong! I updated this part based on the comments.

**Review by XudongLiuHarold — APPROVED** [review]
LGTM

**Review by wjun — APPROVED** [review]
LGTM

**Pull request merged**